hail/src/main/scala/is/hail/methods/LocalLDPrune.scala

```diff
@@ -22,62 +22,69 @@ object BitPackedVector {
   final val BITS_PER_PACK: Int = 2 * GENOTYPES_PER_PACK
 }
 
-private case class BitPackedVectorLoader(fullRowPType: PStruct, callField: String, nSamples: Int) {
+class BitPackedVectorBuilder(nSamples: Int) {
   require(nSamples >= 0)
-  val PField(_, locusType: PLocus, locusIdx) = fullRowPType.fieldByName("locus")
-  val PField(_, allelesType: PArray, allelesIdx) = fullRowPType.fieldByName("alleles")
-  val alleleType = allelesType.elementType.asInstanceOf[PString]
-  val hcView = new HardCallView(fullRowPType, callField)
+  private var idx = 0
+  private var nMissing = 0
+  private var gtSum = 0
+  private var gtSumSq = 0
+
+  private val nPacks = (nSamples - 1) / GENOTYPES_PER_PACK + 1
+  private val packs = new LongArrayBuilder(nPacks)
+  private var pack = 0L
+  private var packOffset = BITS_PER_PACK - 2
+
+  def reset(): Unit = {
+    idx = 0
+    nMissing = 0
+    gtSum = 0
+    gtSumSq = 0
+    packs.clear()
+    pack = 0L
+    packOffset = BITS_PER_PACK - 2
+  }
 
-  def load(ptr: Long): Option[BitPackedVector] = {
-    require(nSamples >= 0)
-    hcView.set(ptr)
+  def addGT(call: Int): Unit = {
+    require(idx < nSamples)
+    val gt = Call.nNonRefAlleles(call)
```

Comment:
This should probably throw an exception on non-diploid calls. Won't be semantically correct if it's not.

Reply:
Ok